lib/auth/auth.go

```diff
@@ -2377,6 +2381,26 @@ func (a *Server) isMFARequired(ctx context.Context, checker services.AccessCheck
 	if !errors.Is(noMFAAccessErr, services.ErrSessionMFARequired) {
 		if trace.IsAccessDenied(noMFAAccessErr) {
 			log.Infof("Access to resource denied: %v", noMFAAccessErr)
+
+			// Emit the failure to the audit log.
+			user := ClientUsername(ctx)
+			if err := a.emitter.EmitAuditEvent(ctx, &apievents.AuthAttempt{
+				Metadata: apievents.Metadata{
+					Type: events.AuthAttemptEvent,
+					Code: events.AuthAttemptFailureCode,
+				},
+				UserMetadata: apievents.UserMetadata{
+					User:  user,
+					Login: noMFAAccessErrTarget,
```

Comment:
Question for reviewers, should `Impersonator: ClientImpersonator(ctx),` be set in the user metadata? I see that used in some audit events but not others.

Reply:
I would include it since it should be useful wherever impersonation might take place.

Reply:
Works for me, added it